subiquitycore/controllers/network.py

```diff
@@ -87,13 +87,31 @@ def route_change(self, action, data):
             return
         if data['table'] != 254:
             return
-        ifindex = data['ifindex']
-        if action == "NEW" or action == "CHANGE":
-            self.default_routes.add(ifindex)
-        elif action == "DEL" and ifindex in self.default_routes:
-            self.default_routes.remove(ifindex)
-        log.debug('default routes %s', self.default_routes)
-        self.controller.update_default_routes(self.default_routes)
+        self.probe_default_routes()
+        self.controller.update_default_routes(self.has_default_routes)
+
+    def _analyze_routes(self, routes):
+        return any(route['table'] == 254 and not route['dst']
+                   for route in routes)
```

Comment:
I think we should consider rejecting routes with metrics > 20000 here, to filter out connections that have not passed NM's connectivity check (unless there is a expectation that such routes might be created by some other mechanism? It seems very unlikely in the context of subiquity though?)

Reply:
I think we should, yes.  These high-metric routes can stick around.  I think it sensible to filter them.  Implemented.